core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -182,12 +180,12 @@ public void registerStream(MovAvgModelStreams.Stream stream) {
         MovAvgModelStreams.registerStream(stream);
     }
 
-    public void registerHighlighter(Class<? extends Highlighter> clazz) {
-        highlighters.add(clazz);
+    public void registerHighlighter(String key, Class<? extends Highlighter> clazz) {
```

Comment:
Can we do away with these entirely and just let plugins work with instances of ExtensionPoint?